ext/openssl/ossl_ssl.c

```diff
@@ -441,6 +441,60 @@ ossl_sslctx_session_new_cb(SSL *ssl, SSL_SESSION *sess)
     return 0;
 }
 
+#if OPENSSL_VERSION_NUMBER >= 0x10101000 && !defined(LIBRESSL_VERSION_NUMBER)
+/*
+ * It is only compatible with OpenSSL >= 1.1.1. Even if LibreSSL implements
+ * SSL_CTX_set_keylog_callback() from v3.4.2, it does nothing (see
+ * https://github.com/libressl-portable/openbsd/commit/648d39f0f035835d0653342d139883b9661e9cb6).
+ */
+
+static VALUE
+ossl_call_keylog_cb(VALUE ary)
+{
+    VALUE ssl_obj, sslctx_obj, cb;
+
+    Check_Type(ary, T_ARRAY);
+    ssl_obj = rb_ary_entry(ary, 0);
+    sslctx_obj = rb_attr_get(ssl_obj, id_i_context);
+
+    cb = rb_attr_get(sslctx_obj, id_i_keylog_cb);
+    if (NIL_P(cb)) return Qnil;
+
+    return rb_funcallv(cb, id_call, 1, &ary);
+}
+
+static VALUE
+rb_str_new_cstr_stub(VALUE str)
+{
+    return rb_str_new_cstr((const char *)str);
+}
```

Comment:
I had to use a stub to avoid some compilation warnings when using `rb_protect()`. Apparently, the first argument has to be a pointer to a function accepting an argument of type `VALUE` and using `rb_str_new_cstr()` directly was an issue:
```
VALUE rb_protect(VALUE (*)(VALUE), VALUE, int*);
```

Please, let me know if there is better way to do this.